Added back-end functionality to send location to frontend

marchdan opened the pull request:
White-listed location and added location column to the data base.

Bad-Science commented on db/migrate/20180228222337_add_periods_location_to_sections.rb:
```diff
@@ -0,0 +1,5 @@
+class AddPeriodsLocationToSections < ActiveRecord::Migration[5.1]
+  def change
+    add_column :sections, :periods_location, :string, array: true, default: []
```
can we add a `null: false` here? Then be sure to run the migration again.

marchdan replied:
Sure thing. Would I just do this on my local machine, then re-push to my fork?

Bad-Science replied:
yup!

marchdan commented:
Commit specifications:
white-listed location in app/consumers/application_consumer.rb

add periods_location to periods_info, added a periods_location check in def periods_changed?, and added periods_location to the periods_hash_to_array method, in app/module/section.rb

Added locations to backend response to frontend in _sections & index.json.jbuilder

added a periods_location migration in db/migrate, entitled:
20180228222337_add_periods_location_to_sections.rb

Ran a migration which updated db/schema.rb

Bad-Science commented:
LGTM, thanks again @marchdan !
